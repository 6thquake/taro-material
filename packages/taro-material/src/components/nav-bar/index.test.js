import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'

import AtNavBar from '../../../.temp/components/nav-bar/index'

describe('AtNavBar Snap', () => {
  it('render initial AtNavBar', () => {
    const component = renderToString(<AtNavBar />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props className', () => {
    const component = renderToString(<AtNavBar className='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props customStyle', () => {
    const component = renderToString(<AtNavBar customStyle='color:red;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props fixed', () => {
    const component = renderToString(<AtNavBar fixed />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props color', () => {
    const component = renderToString(<AtNavBar color='#fff' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props leftIconType', () => {
    const component = renderToString(<AtNavBar leftIconType='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props leftText', () => {
    const component = renderToString(<AtNavBar leftText='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props title', () => {
    const component = renderToString(<AtNavBar title='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props leftText', () => {
    const component = renderToString(<AtNavBar leftText='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props rightFirstIconType', () => {
    const component = renderToString(<AtNavBar rightFirstIconType='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtNavBar -- props rightSecondIconType', () => {
    const component = renderToString(<AtNavBar rightSecondIconType='test' />)
    expect(component).toMatchSnapshot()
  })
})
