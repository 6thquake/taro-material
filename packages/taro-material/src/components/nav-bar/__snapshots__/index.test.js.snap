// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtNavBar Snap render AtNavBar -- props className 1`] = `"<div class=\\"at-nav-bar test\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props color 1`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #fff;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #fff;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #fff;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props fixed 1`] = `"<div class=\\"at-nav-bar at-nav-bar--fixed\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props leftIconType 1`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-test\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props leftText 1`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\">test</span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props leftText 2`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\">test</span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props rightFirstIconType 1`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-test\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props rightSecondIconType 1`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-test\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render AtNavBar -- props title 1`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\">test</div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;

exports[`AtNavBar Snap render initial AtNavBar 1`] = `"<div class=\\"at-nav-bar\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__left_view\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span><span class=\\"taro-text at-nav-bar__text\\"></span></div><div class=\\"at-nav-bar__title\\"></div><div class=\\"at-nav-bar__right_view\\"><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div><div style=\\"color: #6190E8;\\" class=\\"at-nav-bar__container at-nav-bar__container--hide\\"><span style=\\"font-size:24px;color:;\\" class=\\"taro-text at-icon at-icon-\\"></span></div></div></div>"`;
